pkg/hns/endpoint_windows.go

```diff
@@ -40,6 +52,86 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) (*hcsshim.HNSEndpoint, error) {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
+	if hnsEndpoint != nil && err != nil {
+		if hnsEndpoint.VirtualNetwork != epInfo.NetworkId {
+			_, err = hnsEndpoint.Delete()
+			if err != nil {
+				return nil, errors.Annotatef(err, "Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
+			}
+		}
+	} else {
```

Comment:
What happens if hnsEndpoint != nil and error == nil? Does this now overwrite the returned hnsEndpoint?

Reply:
This is fixed in my latest commit